Require image for ARM64 architecture

@a364176773 ,
I am trying to use the **seata/seata** docker image on the **arm64** platform but it seems it is not available for arm64.
I have tried building the image on the arm64 platform using `docker build -t <image_name .>` command, but it is failing with below error on arm64.
```
Step 5/9 : ADD target/ /seata-server/target
ADD failed: file not found in build context or excluded by .dockerignore: stat target/: file does not exist
```

Before using `docker build -t <image_name .>` command, I had tried to build **Uber Jar** through `mvn clean install` but it's falling with below error.
```
[ERROR] Failed to execute goal org.xolstice.maven.plugins:protobuf-maven-plugin:0.5.0:compile (default) on project seata-serializer-protobuf: Missing:
[ERROR] ----------
[ERROR] 1) com.google.protobuf:protoc:exe:linux-aarch_64:3.3.0
[ERROR]
[ERROR]   Try downloading the file manually from the project website.
[ERROR]
[ERROR]   Then, install it using the command:
[ERROR]       mvn install:install-file -DgroupId=com.google.protobuf -DartifactId=protoc -Dversion=3.3.0 -Dclassifier=linux-aarch_64 -Dpackaging=exe -Dfile=/path/to/file
[ERROR]
[ERROR]   Alternatively, if you host your own repository you can deploy the file there:
[ERROR]       mvn deploy:deploy-file -DgroupId=com.google.protobuf -DartifactId=protoc -Dversion=3.3.0 -Dclassifier=linux-aarch_64 -Dpackaging=exe -Dfile=/path/to/file -Durl=[url] -DrepositoryId=[id]
[ERROR]
[ERROR]   Path to dependency:
[ERROR]         1) io.seata:seata-serializer-protobuf:jar:1.5.2-SNAPSHOT
[ERROR]         2) com.google.protobuf:protoc:exe:linux-aarch_64:3.3.0
[ERROR]
[ERROR] ----------
[ERROR] 1 required artifact is missing.
[ERROR]
[ERROR] for artifact:
[ERROR]   io.seata:seata-serializer-protobuf:jar:1.5.2-SNAPSHOT
[ERROR]
[ERROR] from the specified remote repositories:
[ERROR]   central (https://repo.maven.apache.org/maven2, releases=true, snapshots=false)
[ERROR]
[ERROR] -> [Help 1]
[ERROR]
[ERROR] To see the full stack trace of the errors, re-run Maven with the -e switch.
[ERROR] Re-run Maven using the -X switch to enable full debug logging.
[ERROR]
[ERROR] For more information about the errors and possible solutions, please read the following articles:
[ERROR] [Help 1] http://cwiki.apache.org/confluence/display/MAVEN/MojoExecutionException
[ERROR]
[ERROR] After correcting the problems, you can resume the build with the command
[ERROR]   mvn <args> -rf :seata-serializer-protobuf
```

It will be very helpful if the **seata** image is released for the ARM64 platform. If required, I am happy to contribute. But as a start, can you please provide me with some pointers on the above mvn related issue ?

[@a364176773](https://github.com/a364176773) 

 
seata binary is not available for arm64 platform. 
I have tried to built the arm binary for this repository using `mvn clean install `
but, I have faced below error. 

```
[ERROR] Failed to execute goal org.apache.maven.plugins:maven-surefire-plugin:2.19.1:test (default-test) on project seata-common: There are test failures.
[ERROR]
[ERROR] Please refer to /home/abhay/seata/common/target/surefire-reports for the individual test results.
[ERROR] -> [Help 1]
[ERROR]
[ERROR] To see the full stack trace of the errors, re-run Maven with the -e switch.
[ERROR] Re-run Maven using the -X switch to enable full debug logging.
[ERROR]
[ERROR] For more information about the errors and possible solutions, please read the following articles:
[ERROR] [Help 1] http://cwiki.apache.org/confluence/display/MAVEN/MojoFailureException
[ERROR]
[ERROR] After correcting the problems, you can resume the build with the command
[ERROR]   mvn <args> -rf :seata-common
 
```

 After that, tried using `mvn clean compile -Dcheckstyle.skip=true`. It ran successfully but I'm not able to find the exact location of the binaries. I am not sure that binaries are built or not. 

It will be very helpful if you share some pointer on it how can we build and release arm64 binaries for this repo. If required, I am happy to help. Also please have a look on  [this](https://github.com/seata/seata/issues/4660#issuecomment-1154695406) 
Thanks in advance.

--- a/changes/en-us/develop.md
+++ b/changes/en-us/develop.md
@@ -3,6 +3,7 @@ Add changes here for all PR submitted to the develop branch.
 <!-- Please add the `changes` to the following location(feature/bugfix/optimize/test) based on the type of PR -->
 
 ### feature:
+- [[#4802](https://github.com/seata/seata/pull/4802)] dockerfile support arm64
 
 
 ### bugfix:

--- a/changes/zh-cn/develop.md
+++ b/changes/zh-cn/develop.md
@@ -3,6 +3,7 @@
 <!-- 请根据PR的类型添加 `变更记录` 到以下对应位置(feature/bugfix/optimize/test) 下 -->
 
 ### feature：
+- [[#4802](https://github.com/seata/seata/pull/4802)] dockerfile 支持 arm64
 
 
 ### bugfix：

--- a/distribution/Dockerfile
+++ b/distribution/Dockerfile
@@ -13,7 +13,6 @@ ADD target/ /seata-server/target
 ADD lib/ /seata-server/lib
 ADD conf/ /seata-server/conf
 ADD LICENSE /seata-server/LICENSE
-ADD logs /seata-server/logs
 
 # set extra environment
 ENV EXTRA_JVM_ARGUMENTS="-Djava.security.egd=file:/dev/./urandom -server -Xss512k -XX:+UnlockExperimentalVMOptions -XX:+UseContainerSupport XX:SurvivorRatio=10 -XX:MetaspaceSize=128m -XX:MaxMetaspaceSize=256m -XX:MaxDirectMemorySize=1024m -XX:-OmitStackTraceInFastThrow -XX:-UseAdaptiveSizePolicy -XX:+HeapDumpOnOutOfMemoryError -XX:+DisableExplicitGC -Xloggc:/var/log/seata_gc.log -verbose:gc -Dio.netty.leakDetectionLevel=advanced"

--- a/distribution/Dockerfile
+++ b/distribution/Dockerfile
@@ -25,6 +25,5 @@ ADD conf/ /seata-server/conf
 ADD LICENSE /seata-server/LICENSE
 
 # set extra environment
-ENV EXTRA_JVM_ARGUMENTS="-Djava.security.egd=file:/dev/./urandom -server -Xss512k -XX:+UnlockExperimentalVMOptions -XX:+UseContainerSupport XX:SurvivorRatio=10 -XX:MetaspaceSize=128m -XX:MaxMetaspaceSize=256m -XX:MaxDirectMemorySize=1024m -XX:-OmitStackTraceInFastThrow -XX:-UseAdaptiveSizePolicy -XX:+HeapDumpOnOutOfMemoryError -XX:+DisableExplicitGC -Xloggc:/var/log/seata_gc.log -verbose:gc -Dio.netty.leakDetectionLevel=advanced"
 ENV LOADER_PATH="/seata-server/lib"
 CMD ["bash","-c","/seata-server/bin/seata-server.sh && tail -f /dev/null"]

--- a/distribution/Dockerfile
+++ b/distribution/Dockerfile
@@ -1,5 +1,15 @@
+# the Dockerfile support x86 & arrch64
+#
+# build：
+# 1. mvn -Prelease-seata -Dmaven.test.skip=true clean install -U
+# 2. cd distribution/target/seata-server-xxx/seata/
+# 3. docker build --no-cache --build-arg SEATA_VERSION=1.6.0-SNAPSHOT -t seata-server:1.6.0-dev .
+#
+# run:
+# 1. docker run --name=seata-server -d seata-server:1.6.0-dev
+#
 # https://hub.docker.com/orgs/seataio
-FROM openjdk:8u332-jre-slim
+FROM openjdk:8u332
 
 # set label
 LABEL maintainer="Seata <seata.io>"
@@ -17,4 +27,4 @@ ADD LICENSE /seata-server/LICENSE
 # set extra environment
 ENV EXTRA_JVM_ARGUMENTS="-Djava.security.egd=file:/dev/./urandom -server -Xss512k -XX:+UnlockExperimentalVMOptions -XX:+UseContainerSupport XX:SurvivorRatio=10 -XX:MetaspaceSize=128m -XX:MaxMetaspaceSize=256m -XX:MaxDirectMemorySize=1024m -XX:-OmitStackTraceInFastThrow -XX:-UseAdaptiveSizePolicy -XX:+HeapDumpOnOutOfMemoryError -XX:+DisableExplicitGC -Xloggc:/var/log/seata_gc.log -verbose:gc -Dio.netty.leakDetectionLevel=advanced"
 ENV LOADER_PATH="/seata-server/lib"
-CMD ["bash","/seata-server/bin/seata-server.sh"]
+CMD ["bash","-c","/seata-server/bin/seata-server.sh && tail -f /dev/null"]

--- a/distribution/Dockerfile
+++ b/distribution/Dockerfile
@@ -1,5 +1,5 @@
 # https://hub.docker.com/orgs/seataio
-FROM openjdk:8u232-jre-stretch
+FROM openjdk:8u332-jre-slim
 
 # set label
 LABEL maintainer="Seata <seata.io>"
@@ -8,11 +8,14 @@ WORKDIR /$BASE_DIR
 
 # ADD FORM distribution
 ADD bin/ /seata-server/bin
+ADD ext/ /seata-server/ext
 ADD target/ /seata-server/target
+ADD lib/ /seata-server/lib
 ADD conf/ /seata-server/conf
 ADD LICENSE /seata-server/LICENSE
+ADD logs /seata-server/logs
 
 # set extra environment
-ENV EXTRA_JVM_ARGUMENTS="-Djava.security.egd=file:/dev/./urandom -server -Xss512k -XX:+UnlockExperimentalVMOptions -XX:+UseContainerSupport XX:SurvivorRatio=10 -XX:MetaspaceSize=128m -XX:MaxMetaspaceSize=256m -XX:MaxDirectMemorySize=1024m -XX:-OmitStackTraceInFastThrow -XX:-UseAdaptiveSizePolicy -XX:+HeapDumpOnOutOfMemoryError -XX:+DisableExplicitGC -XX:+CMSParallelRemarkEnabled -XX:+UseCMSInitiatingOccupancyOnly -XX:CMSInitiatingOccupancyFraction=75 -Xloggc:/var/log/seata_gc.log -verbose:gc -Dio.netty.leakDetectionLevel=advanced"
-
-CMD ["sh","/seata-server/bin/seata-server.sh"]
+ENV EXTRA_JVM_ARGUMENTS="-Djava.security.egd=file:/dev/./urandom -server -Xss512k -XX:+UnlockExperimentalVMOptions -XX:+UseContainerSupport XX:SurvivorRatio=10 -XX:MetaspaceSize=128m -XX:MaxMetaspaceSize=256m -XX:MaxDirectMemorySize=1024m -XX:-OmitStackTraceInFastThrow -XX:-UseAdaptiveSizePolicy -XX:+HeapDumpOnOutOfMemoryError -XX:+DisableExplicitGC -Xloggc:/var/log/seata_gc.log -verbose:gc -Dio.netty.leakDetectionLevel=advanced"
+ENV LOADER_PATH="/seata-server/lib"
+CMD ["bash","/seata-server/bin/seata-server.sh"]

--- a/distribution/bin/seata-server.sh
+++ b/distribution/bin/seata-server.sh
@@ -126,8 +126,9 @@ JVM_XSS=$JVM_XSS
 JVM_MetaspaceSize=$JVM_MetaspaceSize
 JVM_MaxMetaspaceSize=$JVM_MaxMetaspaceSize
 JVM_MaxDirectMemorySize=$JVM_MaxDirectMemorySize
-JAVA_OPT="${JAVA_OPT} -server -Dloader.path=../lib -Xmx${JVM_XMX:="2048m"} -Xms${JVM_XMS:="2048m"} -Xmn${JVM_XMN:="1024m"} -Xss${JVM_XSS:="512k"} -XX:SurvivorRatio=10 -XX:MetaspaceSize=${JVM_MetaspaceSize:="128m"} -XX:MaxMetaspaceSize=${JVM_MaxMetaspaceSize:="256m"} -XX:MaxDirectMemorySize=${JVM_MaxDirectMemorySize:=1024m} -XX:-OmitStackTraceInFastThrow -XX:-UseAdaptiveSizePolicy"
-JAVA_OPT="${JAVA_OPT} -XX:+HeapDumpOnOutOfMemoryError -XX:HeapDumpPath=${BASEDIR}/logs/java_heapdump.hprof -XX:+DisableExplicitGC -XX:+CMSParallelRemarkEnabled -XX:+UseCMSInitiatingOccupancyOnly -XX:CMSInitiatingOccupancyFraction=75"
+LOADER_PATH=$LOADER_PATH
+JAVA_OPT="${JAVA_OPT} -server -Dloader.path=${LOADER_PATH:="../lib"} -Xmx${JVM_XMX:="2048m"} -Xms${JVM_XMS:="2048m"} -Xmn${JVM_XMN:="1024m"} -Xss${JVM_XSS:="512k"} -XX:SurvivorRatio=10 -XX:MetaspaceSize=${JVM_MetaspaceSize:="128m"} -XX:MaxMetaspaceSize=${JVM_MaxMetaspaceSize:="256m"} -XX:MaxDirectMemorySize=${JVM_MaxDirectMemorySize:=1024m} -XX:-OmitStackTraceInFastThrow -XX:-UseAdaptiveSizePolicy"
+JAVA_OPT="${JAVA_OPT} -XX:+HeapDumpOnOutOfMemoryError -XX:HeapDumpPath=${BASEDIR}/logs/java_heapdump.hprof -XX:+DisableExplicitGC "
 
 JAVA_MAJOR_VERSION=$($JAVACMD -version 2>&1 | sed '1!d' | sed -e 's/"//g' | awk '{print $3}' | awk -F '.' '{print $2}')
 if [[ "$JAVA_MAJOR_VERSION" -ge "9" ]] ; then

--- a/distribution/bin/seata-server.sh
+++ b/distribution/bin/seata-server.sh
@@ -151,10 +151,5 @@ CMD_LINE_ARGS=$@
 
 # start
 echo "$JAVACMD ${JAVA_OPT} ${CMD_LINE_ARGS}" > ${BASEDIR}/logs/start.out 2>&1 &
-IN_DOCKER_RUNNING=$( mount | grep '/ type' | egrep  '(overlay|aufs)' )
-if [[ "$IN_DOCKER_RUNNING" == "" ]] ; then
-  nohup $JAVACMD ${JAVA_OPT} ${CMD_LINE_ARGS} >> ${BASEDIR}/logs/start.out 2>&1 &
-else
-  $JAVACMD ${JAVA_OPT} ${CMD_LINE_ARGS}
-fi
+nohup $JAVACMD ${JAVA_OPT} ${CMD_LINE_ARGS} >> ${BASEDIR}/logs/start.out 2>&1 &
 echo "seata-server is starting, you can check the ${BASEDIR}/logs/start.out"

--- a/distribution/bin/seata-server.sh
+++ b/distribution/bin/seata-server.sh
@@ -1,4 +1,4 @@
-#!/bin/sh
+#!/bin/bash
 # Copyright 1999-2019 Seata.io Group.
 #
 # Licensed under the Apache License, Version 2.0 (the "License");
@@ -128,7 +128,7 @@ JVM_MaxMetaspaceSize=$JVM_MaxMetaspaceSize
 JVM_MaxDirectMemorySize=$JVM_MaxDirectMemorySize
 LOADER_PATH=$LOADER_PATH
 JAVA_OPT="${JAVA_OPT} -server -Dloader.path=${LOADER_PATH:="../lib"} -Xmx${JVM_XMX:="2048m"} -Xms${JVM_XMS:="2048m"} -Xmn${JVM_XMN:="1024m"} -Xss${JVM_XSS:="512k"} -XX:SurvivorRatio=10 -XX:MetaspaceSize=${JVM_MetaspaceSize:="128m"} -XX:MaxMetaspaceSize=${JVM_MaxMetaspaceSize:="256m"} -XX:MaxDirectMemorySize=${JVM_MaxDirectMemorySize:=1024m} -XX:-OmitStackTraceInFastThrow -XX:-UseAdaptiveSizePolicy"
-JAVA_OPT="${JAVA_OPT} -XX:+HeapDumpOnOutOfMemoryError -XX:HeapDumpPath=${BASEDIR}/logs/java_heapdump.hprof -XX:+DisableExplicitGC "
+JAVA_OPT="${JAVA_OPT} -XX:+HeapDumpOnOutOfMemoryError -XX:HeapDumpPath=${BASEDIR}/logs/java_heapdump.hprof -XX:+DisableExplicitGC -XX:+CMSParallelRemarkEnabled -XX:+UseCMSInitiatingOccupancyOnly -XX:CMSInitiatingOccupancyFraction=75"
 
 JAVA_MAJOR_VERSION=$($JAVACMD -version 2>&1 | sed '1!d' | sed -e 's/"//g' | awk '{print $3}' | awk -F '.' '{print $2}')
 if [[ "$JAVA_MAJOR_VERSION" -ge "9" ]] ; then
@@ -151,5 +151,10 @@ CMD_LINE_ARGS=$@
 
 # start
 echo "$JAVACMD ${JAVA_OPT} ${CMD_LINE_ARGS}" > ${BASEDIR}/logs/start.out 2>&1 &
-nohup $JAVACMD ${JAVA_OPT} ${CMD_LINE_ARGS} >> ${BASEDIR}/logs/start.out 2>&1 &
+IN_DOCKER_RUNNING=$( mount | grep '/ type' | egrep  '(overlay|aufs)' )
+if [[ "$IN_DOCKER_RUNNING" == "" ]] ; then
+  nohup $JAVACMD ${JAVA_OPT} ${CMD_LINE_ARGS} >> ${BASEDIR}/logs/start.out 2>&1 &
+else
+  $JAVACMD ${JAVA_OPT} ${CMD_LINE_ARGS}
+fi
 echo "seata-server is starting, you can check the ${BASEDIR}/logs/start.out"

